src/index.js

```diff
@@ -33,16 +38,28 @@ module.exports = options => ({
   },
   // Fix sourcemaps
   result(stylelintResult, filepath) {
+    const interpolationLines = interpolationLinesMap[filepath] || []
     const lineCorrection = sourceMapsCorrections[filepath]
-    const warnings = stylelintResult.warnings.map(warning =>
-      Object.assign({}, warning, {
-        // Replace "brace" with "backtick" in warnings, e.g.
-        // "Unexpected empty line before closing backtick" (instead of "brace")
-        text: warning.text.replace(/brace/, 'backtick'),
-        line: lineCorrection[warning.line]
-      })
-    )
+    const warnings = stylelintResult.warnings
+      .filter(
+        warning =>
+          // Filter indentation warnings generated by interpolations substitution
+          warning.rule !== 'indentation' ||
+          interpolationLines.indexOf(lineCorrection[warning.line]) < 0
+      )
+      .map(warning =>
+        Object.assign({}, warning, {
+          // Replace "brace" with "backtick" in warnings, e.g.
+          // "Unexpected empty line before closing backtick" (instead of "brace")
+          text: warning.text.replace(/brace/, 'backtick'),
+          line: lineCorrection[warning.line]
+        })
+      )
 
-    return Object.assign({}, stylelintResult, { warnings })
+    return Object.assign({}, stylelintResult, {
+      warnings,
+      // Remember to fix
+      errored: warnings.length ? true : undefined
```

Comment:
This modifying of the stylelintResult definitely makes me a bit nervous in general, but I think your filter is mostly pretty contained so I think I can be okay with it, we need to be really accurate though. Are you sure that this behaviour you wrote here conforms with how Stylelint does it exactly? The only documentation I can find that's the most reliable is these two: https://stylelint.io/user-guide/node-api/ https://stylelint.io/developer-guide/processors/ and this code https://github.com/stylelint/stylelint/blob/master/lib/createStylelintResult.js which may or may not be the object we're passed as I didn't dig deeper into it.

Could you please dig into it deeper and come back with a bit of "proof of correctness" as I could imagine that we may need to check if the rules have `severity: 'error'` or something like that, though the thing is I'm not sure so just want to make sure we don't screw something up here

Reply:
This really makes me unhappy here. I found that `errored` can be `undefined/null/true`. See [stylelint#createStylelintResult.js](https://github.com/stylelint/stylelint/blob/master/lib/createStylelintResult.js#L119).

It is hard to recover `errored` unless authors of `stylelint` can give us a reasonable explanation in [stylelint#3041](https://github.com/stylelint/stylelint/issues/3041).

Reply:
Hmm yes that does seem pretty messy indeed, let's wait a bit and see if you get a reply there, but one thing that I can definitely see from your research ([this part](https://github.com/stylelint/stylelint/blob/abe53c01878e28670540586b27bf001c7316c3fd/lib/utils/report.js#L76) in particular) is that we need to be doing a check here whether any of the warnings have severity level "error", but yeah, maybe we also need to check the null and other stuff.

One way would also be to check whether the `errored` flag is currently set, if it is not we know we don't have to change anything (as we're only removing errors in our modifications so it would never cause an error to show up, only to be removed), if it is set to true though, then check if any of the remaining warnings have severity level "error", and if they don't change `errored` to `undefined` I guess right? It should be `undefined` not `false` according to your research yeah?

I think this would handle the case where it's null etc. as well, what do you think?

Reply:
Right, I should check severity levels. Good news is that ignored files will not invoke our processor, so we can do not consider the `null` case.

Reply:
Nice :). 

Yeah so I think the method I proposed above should work yeah? Don't do anything if it's falsy, if it is truthy check if any warnings with severity level error, and if there aren't any change it to `undefined`.